sdk/ai/azure-ai-generative/azure/ai/generative/evaluate/metrics/_parsers.py

```diff
@@ -0,0 +1,47 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import logging
+import re
+from json import JSONDecodeError
+
+import numpy as np
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ScoreReasonParser(object):
+
+    @staticmethod
+    def parse(value, metric):
+        try:
+            value_json = json.loads(value)
+            score = value_json.get("score")
+            reason = value_json.get("reason")
+        except JSONDecodeError as json_parse_error:
+            LOGGER.debug(
+                f"Error parsing metric {metric.name} value as returned json from LLM is not a valid json : {value}")
```

Comment:
I remember openai has a new feature to enforce the response to be a valid json, are we using that feature? Here's a link. https://community.openai.com/t/how-do-i-use-the-new-json-mode/475890

Reply:
This is not supported on older models for for now would like to go with this approach.

Reply:
By older models what do you mean? Does gpt35 turbo support this?

Reply:
There are specific version of gpt3.5 turbo and possibly gpt-4 too. https://community.openai.com/t/how-do-i-use-the-new-json-mode/475890/2